shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -1043,7 +1043,8 @@ - (BOOL)canBecomeFirstResponder {
 - (BOOL)resignFirstResponder {
   BOOL success = [super resignFirstResponder];
   if (success) {
-    [self.textInputDelegate flutterTextInputViewDidResignFirstResponder:self];
+    [self.textInputDelegate flutterTextInputViewDidResignFirstResponder:self
```

Comment:
nit: Looks like this one will be called when either the Flutter framework or UIKit tells the input view to give up its first responder status. It seems redundant for the former since there's no need to inform the Flutter framework when it is the one that initiated the focus change.

Reply:
Yes. When framework close the keyboard first, this will send a msg to framework and will handle return by 

https://github.com/flutter/flutter/blob/659ba386f6d045690f9783c79fe5d320fb7847a1/packages/flutter/lib/src/services/text_input.dart#L1788

And only when platform starts to unfocus, the `_currentConnection!._client.connectionClosed();` will be called. 

So I think this cost can be acceptable?

Reply:
Yeah that makes sense the old connection is already torn down by the framework so messages associated with the old client id will be thrown away.